lib/grpc/codec.ex

```diff
@@ -7,4 +7,6 @@ defmodule GRPC.Codec do
   @callback name() :: String.t()
   @callback encode(any) :: binary
   @callback decode(any, module :: atom) :: any
+  @callback prepare_decode(binary) :: binary
```

Comment:
Not sure if prepare_decode is a good name. This is still part of the decoding process, right?

Reply:
Perhaps something more akin to "decode_channel" as in channel-encoding vs application-encoding?

I don't really have a better suggestion. If we can't find any, let's go with `prepare_decode`

Reply:
Perhaps since we have `pack_encoded`, we could have `pack_for_channel` and `unpack_from_channel`

Reply:
Also, I think a `@doc` would be desirable for both of the functions

Reply:
These could be optional callbacks and we could just check if they are defined by the codec instead of having to define no-op impls

Reply:
Agree! Let's call them `pack_for_channel`, `unpack_for_channel`.